brave/browser-laptop
# Re-block running insecure content

---
1. Add an option to re-block running insecure content
2. Limit the effect of running insecure content in regular/private frame scope

fix #3770, #3793 

partially fix #3795 (only for runInsecureContent)

Auditors: @diracdeltas

Test Plan:
For re-block option:
1. Go to `https://mixed-script.badssl.com/`
2. Click urlIcon and click `Load Unsafe Scripts`
3. Background should be red
4. Click urlIcon and click `Stop Loading Unsafe Scripts`
5. Background should be grey

For regular frame:
1. Go to https://mixed-script.badssl.com
2. Click urlIcon and click Load Unsafe Scripts
3. Background should be red
4. Go to https://mixed-script.badssl.com by regular new tab
5. Background should be grey
6. Go to https://mixed-script.badssl.com by private new tab
7. Background should be grey
8. Go to https://mixed-script.badssl.com by session new tab
9. Background should be grey

~~For private frame:
1. Go to https://mixed-script.badssl.com by private tab
2. Click urlIcon and click Load Unsafe Scripts
3. Background should be red
4. Go to https://mixed-script.badssl.com by regular new tab
5. Background should be grey
6. Go to https://mixed-script.badssl.com by private new tab
7. Background should be grey
8. Go to https://mixed-script.badssl.com by session new tab
9. Background should be grey~~
